tests/addnode.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import e2e_args
+import infra.ccf
+import infra.proc
+import json
+
+import logging
+import time
+
+from loguru import logger as LOG
+
+
+def create_and_add_node(
+    network, args, node_id, member_host, member_port, should_succeed=True
+):
+    dict_args = vars(args)
+    forwarded_args = {
+        arg: dict_args[arg] for arg in infra.ccf.Network.node_args_to_forward
+    }
+    node_status = args.node_status or "pending"
+    new_node = network.create_node(node_id, "localhost")
+    new_node.start(
+        lib_name=args.package,
+        node_status=node_status,
+        workspace=args.workspace,
+        label=args.label,
+        **forwarded_args
+    )
+    new_node_info = new_node.remote.info()
+    new_cert_path = "{}/{}".format(
```

Comment:
We could add a new function to `Node` that returns the node's certificate path.

Reply:
Starring at this for longer, `new_node_info` should already contain the raw certificate and quote. So if we pass a `node.json` file to the `memberclient` instead, this code should get simpler.